src/main/io/gps.c

```diff
@@ -641,7 +641,8 @@ static uint32_t grab_fields(char *src, uint8_t mult)
 {                               // convert string to uint32
     uint32_t i;
     uint32_t tmp = 0;
-    for (i = 0; src[i] != 0; i++) {
+    int isneg = src[0] == '-';
```

Comment:
it is probably OK (and cleaner) to ignore `-` on first parsing (for (i = 0; ... ) and use it when returning value